typings/n-express.d.ts

```diff
@@ -0,0 +1,80 @@
+import Metrics from '@financial-times/n-metrics';
+import Express from 'express';
+import http from 'http';
+import https from 'https';
+import { cacheHeaders } from '../src/middleware/cache';
+export * from './metrics';
+
+export type CacheHeaders = typeof cacheHeaders;
+
+export interface Request extends Express.Request {}
+
+export interface Response extends Express.Response, CacheHeaders {
+	unvary: (...args: string[]) => void;
+	unvaryAll: () => void;
+
+	// Backwards-compatible uber-camel-cased names
+	unVary: unvary;
+	unVaryAll: unvaryAll;
+}
+
+export interface NextFunction extends Express.NextFunction {}
+
+export type Callback = (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => void;
```

Comment:
**thought**: it doesn't look like this type is used for express's `app.get` etc, so apps will only see the base express `Request` etc there, do you plan to address that?

Reply:
oh i guess that's what the TODO comment below is talking about

Reply:
> oh i guess that's what the TODO comment below is talking about

Yeah exactly, I was hoping there would be some way to map the function signatures to use our `NExpress` type instead but no dice as far as I can tell. Will probably have to copy and paste the `express` definition instead but that's a bit messy and fragile so I think we should put it off until we see a case where it's actually needed first.